ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs

```diff
@@ -188,15 +189,54 @@ newFreshIdSupply _ = do
     return $ FreshIdSupply { getFreshId }
 
 
-newMutableConnState :: MonadSTM m
-                    => FreshIdSupply m
+newMutableConnState :: forall peerAddr handle handleError version m.
+                      ( MonadTraceSTM m
+                      , Typeable peerAddr
+                      )
+                    => peerAddr
+                    -> FreshIdSupply m
                     -> ConnectionState peerAddr handle handleError
                                        version m
                     -> STM m (MutableConnState peerAddr handle handleError
                                                version m)
-newMutableConnState freshIdSupply connState = do
+newMutableConnState peerAddr freshIdSupply connState = do
       connStateId <- getFreshId freshIdSupply
       connVar <- newTVar connState
+      -- This tracing is a no op in IO.
+      --
+      -- We need this for IOSimPOR testing of connection manager state
+      -- transition tests. It can happen that the transitions happen
+      -- correctly but IOSimPOR reorders the threads that log the transitions.
+      -- This is a false positive and we don't want that to happen.
+      --
+      -- The simplest way to do so is to leverage the `traceTVar` IOSim
+      -- capabilities. These trace messages won't be reordered by IOSimPOR
+      -- since these happen atomically in STM.
+      --
+      -- Another thing to note is that this trace differs from the IO one in
+      -- the fact that all connections terminate with a trace to
+      -- 'UnknownConnectionSt', since we can't do that here we limit ourselves
+      -- to 'TerminatedSt'.
```

Comment:
You think so? I think it is in the right place, because a person reading the code might be wondering why this snippet of code is here and why we disregard the `UnknownConnectionSt` transition. I can put it in the testsuite as well

Reply:
I think so.  This is only important in the test, not for IO.